assets/src/edit-story/components/canvas/carousel/index.js

```diff
@@ -216,13 +225,18 @@ function Carousel() {
     }) => ({ pages, currentPageId, setCurrentPage, arrangePage })
   );
   const { isRTL } = useConfig();
+  const { showSafeZone, setShowSafeZone } = useCanvas(
+    ({ state: { showSafeZone }, actions: { setShowSafeZone } }) => ({
+      showSafeZone,
+      setShowSafeZone,
+    })
+  );
```

Comment:
I wonder if we can set up our useContextSelector pattern to just accept some sort of shorthand for this frequently occurring case where we're just pulling selected properties off of the context.
```
  const { showSafeZone, setShowSafeZone } = useCanvas([['showSafeZone'], ['setShowSafeZone']]);
// or something more like dlv
  const { showSafeZone, setShowSafeZone } = useCanvas('state.showSafeZone', 'actions.setShowSafeZone');
```

Or maybe we just make it a pattern to statically define these outside the component definition:
```
const selector = ({ state: { showSafeZone }, actions: { setShowSafeZone } }) => ({
    showSafeZone,
    setShowSafeZone,
});
...
function Carousel() {
  ...
  const { showSafeZone, setShowSafeZone } = useCanvas(selector);
  ...
}
```

Reply:
I like the idea of defining the selector outside the component definition a lot.  Going to add that now.